lib/resources/oracledb_session.rb

```diff
@@ -1,41 +1,139 @@
 # encoding: utf-8
 # author: Nolan Davidson
+# author: Christoph Hartmann
+# author: Dominik Richter
+
+require 'hashie/mash'
+require 'utils/database_helpers'
+require 'htmlentities'
+require 'rexml/document'
+require 'csv'
 
 module Inspec::Resources
+  # STABILITY: Experimental
+  # This resouce needs further testing and refinement
+  #
   class OracledbSession < Inspec.resource(1)
     name 'oracledb_session'
     desc 'Use the oracledb_session InSpec resource to test commands against an Oracle database'
     example "
       sql = oracledb_session(user: 'my_user', pass: 'password')
-      describe sql.query('SELECT NAME FROM v$database;') do
-        its('stdout') { should_not match(/test/) }
+      describe sql.query(\"SELECT UPPER(VALUE) AS VALUE FROM V$PARAMETER WHERE UPPER(NAME)='AUDIT_SYS_OPERATIONS'\").row(0).column('value') do
+        its('value') { should eq 'TRUE' }
       end
     "
 
-    attr_reader :user, :pass, :host, :sid, :sqlplus_bin
-
+    attr_reader :user, :password, :host, :service
     def initialize(opts = {})
       @user = opts[:user]
-      @pass = opts[:pass]
+      @password = opts[:password] || opts[:pass]
+      if opts[:pass]
+        warn '[DEPRECATED] use `password` option to supply password instead of `pass`'
+      end
+
       @host = opts[:host] || 'localhost'
-      @sid = opts[:sid]
+      @port = opts[:port] || '1521'
+      @service = opts[:service]
+
+      # we prefer sqlci although it is way slower than sqlplus, but it understands csv properly
+      @sqlcl_bin = 'sql'
       @sqlplus_bin = opts[:sqlplus_bin] || 'sqlplus'
-      return skip_resource("Can't run Oracle checks without authentication") if @user.nil? or @pass.nil?
+
+      return skip_resource "Can't run Oracle checks without authentication" if @user.nil? || @password.nil?
+      return skip_resource 'You must provide at least an SID and/or a Service Name for the session' if @service.nil?
```

Comment:
We should eliminate the reference to "SID" from this skip message since we only support service name.